app/javascript/components/Modal.jsx

```diff
@@ -0,0 +1,29 @@
+import PropTypes from "prop-types";
+import React from "react";
+
+class Modal extends React.Component {
```

Comment:
Sorry! In addition, you should also use an existing modal library. This is because the way you implement it currently, you can still access elements on the screen behind the modal. Instead, we need to prevent the user from making these changes. You can implement an existing modal library such as react-modal or ReactBootstrap.Modal